core/App.php

```diff
@@ -287,11 +287,27 @@ public static function checkout_process($orderId){
                     }
                 }
 
-
+                $expiry = Settings::get('expiry');
+                $auth = array(
+                    'access_token' => Settings::get('access_token'),
+                    'refresh_token' => Settings::get('refresh_token')
+                );
+                if ($expiry !== null && ($expiry - time() <  (60*60*24)))

```

Comment:
Presumably public access is only adding these subscribers right - and all others should be within admin / connect - right? this is similar code with the admin, so optionally to factor it into a single method that accessed from both. NB: expiry will never be null if we hit this page, right?

Reply:
I'll put it in one method. And yeah expiry shouldn't be null at this point

Reply:
Actually it is sorta different since one is in a view page and the other is in a class object. And the one in the view page is also tied to an if/else statement